src/main/java/org/testng/TestRunner.java

```diff
@@ -784,20 +786,27 @@ private void privateRun(XmlTest xmlTest) {
    * Apply the method interceptor (if applicable) to the list of methods.
    */
   private ITestNGMethod[] intercept(ITestNGMethod[] methods) {
-    if (m_methodInterceptor == null) return methods;
+    IMethodInstance[] methodInstances = methodsToMethodInstances(Arrays.asList(methods));
 
-    IMethodInstance[] instances = methodsToMethodInstances(Arrays.asList(methods));
+    // if there're more than 1 interceptor then it will add the first one (InstanceOrderingInterceptor or PreserveOrderInterceptor) at the end.
```

Comment:
Or you may store the OrderingInterceptor outside the list, and apply it after all others.